pkgs/development/tools/confluent-cli/default.nix

```diff
@@ -2,17 +2,29 @@
 
 stdenv.mkDerivation rec {
   pname = "confluent-cli";
-  version = "3.17.0";
+  version = "3.37.0";
 
   # To get the latest version:
   # curl -L https://cnfl.io/cli | sh -s -- -l | grep -v latest | sort -V | tail -n1
-  src = fetchurl (if stdenv.hostPlatform.isDarwin then {
+  src = fetchurl (
+    if stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64 then {
       url = "https://s3-us-west-2.amazonaws.com/confluent.cloud/confluent-cli/archives/${version}/confluent_${version}_darwin_amd64.tar.gz";
-      sha256 = "03104736f65591a5be9536424460d9b8c8fc8ac8b5eb646e832371397aaf7cef";
+      hash = "sha256-6i5Z3m0gMzPuqWm/SJOuxjO3ioh/Uhk3A9uykksYfPo=";
+
+    } else if stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isx86_64 then {
+      url = "https://s3-us-west-2.amazonaws.com/confluent.cloud/confluent-cli/archives/${version}/confluent_${version}_darwin_amd64.tar.gz";
+      hash = "sha256-gEQBxChsM5CXFRsWBVVcQ88xQ2N4lqkIxHfZKPEMlOY";
+
+    } else if stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isAarch64 then {
+      url = "https://s3-us-west-2.amazonaws.com/confluent.cloud/confluent-cli/archives/${version}/confluent_${version}_linux_arm64.tar.gz";
+      hash = "sha256-lj7i7oQzX1AfhYfrXDiOjz1/EV4y3/CI4MyPKzNGcss=";
+
+    # defaults to linux amd64
     } else {
       url = "https://s3-us-west-2.amazonaws.com/confluent.cloud/confluent-cli/archives/${version}/confluent_${version}_linux_amd64.tar.gz";
-      sha256 = "3243beca4fefd49cf067f9a4df4f5072a8ac5dac91638e9f10ef0b0544d30445";
-    });
+      hash = "sha256-vJB/0odVA86fZtRh/Cg5KPD8q8CQFENlRzjpI41UOc8=";
+    }
+  );
```

Comment:
Something like this would be much better, as it doesn't assume a default platform and doesn't have duplicated conditions:

https://github.com/NixOS/nixpkgs/blob/a7e91f0509c6b6d2a7a37aac6622f7469590234d/pkgs/development/compilers/julia/1.9-bin.nix#L29-L46

Reply:
Oooh, agreed! I didn’t know a lot of those constructs were possible. Happy to incorporate that. Thanks for the suggestion!